### DISCUSSION TITLE
Temporary directory, mkdir() permission denied

### DISCUSSION THREAD

**Discussion by ghost** [spatie/laravel-medialibrary#3058] [Q&A]
Hello ! 

The uploading is working well locally, but after pushin my website in production here is the error I get when I try to upload a Media:

`production.ERROR: mkdir(): Permission denied {"userId":12,"exception":"[object] (ErrorException(code: 0): mkdir(): Permission denied at .../vendor/spatie/temporary-directory/src/TemporaryDirectory.php:82)
`

In my composer.json : 
"spatie/laravel-medialibrary": "^10.0.0",
"laravel/nova": "~4.0", 
"ebess/advanced-nova-media-library": "^4.0",

Thank you for your help!

**Comment by parallels999** [comment]
Web user don't have permissions to create folders, temporary-directory creates a temporary folder
[https://github.com/spatie/temporary-directory/src/TemporaryDirectory.php#L81-L83](https://github.com/spatie/temporary-directory/blob/17788feaf04605176824410de1208b818bd4f263/src/TemporaryDirectory.php#L81-L83)

**Reply by ghost** [reply]
Hey, thank you for that precision, what could be the solution then ?

**Comment by jose123v** [comment]
Look at the path permissions, check if the folder exists
